app/models/user.rb

```diff
@@ -43,16 +44,29 @@ def forget
     update_column :remember_digest, nil
   end
 
-  def authenticated? remember_token
-    return false if !remember_token || !remember_digest
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false if !token || !digest
 
-    BCrypt::Password.new(remember_digest)
-                    .is_password? remember_token
+    BCrypt::Password.new(digest).is_password? token
+  end
+
+  def activate
+    update activated: true, activated_at: Time.zone.now
```

Comment:
```suggestion
    update_columns activated: true, activated_at: Time.zone.now
```
Nên dùng thèn này nhé - vì chỗ này ko cần bắt validate hay callback mô